index.bs

```diff
@@ -292,16 +295,46 @@ the PrivacyCG.
 
 * TODO: Define how bounce tracking information is stored; e.g. sites,
         timestamps, etc.
-* TODO: Define how site interaction information is stored.
 * TODO: Define a recurring global timer to run the analyze and delete algorithm.
 
+The user agent holds a <dfn>user activation map</dfn> which is a [=map=] of
+[=site=] [=hosts=] to [=moments=].  The [=moments=] represent the most recent
+[=wall clock=] time at which the user activated a top-level document on the
+associated [=host=].
+
+<p class=note>
+Schemeless site is used as the data structure key because by default cookies
+are sent to both `http://` and `https://` pages on the same domain.
+</p>
+
 <h3 id="bounce-tracking-mitigations-algorithms">Algorithms</h3>
 
 * TODO: Define the steps necessary to detect and store a "bounce".
-* TODO: Define the steps necessary to store user interactions.
 * TODO: Define the steps to analyze information in the data model and delete
         appropriate sites.
 
+<h4 id="bounce-tracking-mitigations-activation-monkey-patch">User Activation Monkey Patch</h4>
+
+<div algorithm>
+
+To <dfn>record a user activation</dfn> given a [=Document=] |document|, perform
+the following steps:
+
+1. If |document|'s [=node navigable=] is null, then abort these steps.
+1. Let |settings| be |document|'s [=relevant settings object=].
+1. Let |origin| be |settings|' [=environment/top-level origin=].
```

Comment:
Oh, I just found https://github.com/whatwg/html/issues/8014, which indicates that we should be moving away from using this, and instead go through the document or window's navigable.